pkg/controllers/routing/ecmp_vip.go

```diff
@@ -141,7 +142,7 @@ func getMissingPrevGen(old, new []string) (withdrawIPs []string) {
 
 // OnServiceDelete handles the service delete updates from the kubernetes API server
 func (nrc *NetworkRoutingController) OnServiceDelete(obj interface{}) {
-	nrc.tryHandleServiceUpdate(obj, "Received event to delete service: %s/%s from watch API")
+	nrc.tryHandleServiceUpdate(obj, "Received event to delete service: %s/%s from watch API", true)
```

Comment:
I couldn't find a reliable way of determining from the `v1core.Service` object if the service was deleted, so sadly resorted to passing a flag down.